oak/common/app_config.cc

```diff
@@ -191,21 +191,21 @@ bool ValidApplicationConfig(const ApplicationConfiguration& config) {
 
         // Track which node.port instances are IN/OUT.
         if (port.type() == Port_Type_IN) {
-          in_ports.insert(fqpn(node.node_name(), port.name()));
+          in_ports[fqpn(node.node_name(), port.name())] = 0;
```

Comment:
Should we maybe have a check if two nodes have the same fqpn?

Reply:
Two distinct nodes must have different node names (checked at l.174 above) so <node, port> name will necessarily be different.